Building and creation of binary release using GitHub Actions

AndrewSav opened the pull request:
Hey, I admit, it's quite a big change, and you may be reluctant to merge it. This is fine by me, just let me know and I'll maintain my separate fork. On the other hand, if you are willing to merge but need things fixed / clarified, please do not hesitate to let me know.

- Implemented build and release creation using GitHub Actions
- Created a build script
- Changed readme format from rst to md
- Re-worked readme to reflect addition of the build script and Github Actions
- Updated versions of dependencies

AndrewSav committed:
- Implemented build and release creation using GitHub Actions
- Create a build script
- Changed readme format from rst to md
- Re-worked readme to refelct addition of the build script and Github Actions
- Updated versions of dependencies

maxwell-k commented:
I like this a lot, thank you. I'd like to go through it in detail. There are a few things I'd like to try out and possibly a tweak or two but I would like to get this merged.

The build below is successful build on AndrewSav's fork:
https://github.com/AndrewSav/ledger-windows-build/runs/590980954

AndrewSav commented:
And the result of the successful build above were uploaded to https://github.com/AndrewSav/ledger-windows-build/releases/tag/v3.1.3 by Github Actions. Please do not hesitate to ask if there are any questions or issues.

AndrewSav committed:
Updated to ledger 3.2.1, latest mpir and mpfr and performed GiHub actions maintenance.
- Removed outdated set-env in actions.
- Updated msbuild-setup action.
- Compensated for BOOST_ROOT removal in virtual environment.
- Added version number to the name of the ledger zip file
- Clarified readme about updating submodules in a repo that was not freshly cloned

AndrewSav commented:
Updated to ledger v3.2.1 https://github.com/AndrewSav/ledger-windows-build/releases/tag/v3.2.1

AndrewSav committed:
updated submodules, include boost build, since removed from Github Virtual Environement

AndrewSav committed:
Update to ledger 3.2.2

AndrewSav committed:
show environment before build in actions
